floodviz/views.py

```diff
@@ -1,7 +1,112 @@
 from flask import request, render_template, abort
+import csv
+import requests
+import yaml
+import os
 
 from . import app
 
+
 @app.route('/home/')
 def home():
-    return render_template('index.html')
\ No newline at end of file
+    return render_template('index.html')
+
+
+# This program requires svgis to be installed (which in turn requires Fiona and GDAL)
+# Takes a list of site ids in the form of "site_ids.csv" (IDs now given in config file)
+# Takes config data in the form of "site_configs.yaml" (also now given in config file)
+# Takes style in the form of "site_styles.css"
+# Adds a city layer in the form of "cities.json"
+# outputs an SVG with ids for each site
+
+siteList = app.config['SITE_IDS']
+
+bounds = app.config['BOUNDS']
+
+# put each site number from the csv into a list
+# with open('static/data/site_ids.csv', newline='') as csvfile:
+#     siteReader = csv.reader(csvfile)
+#     for site in siteReader:
+#         if site[0].isalnum():
+#             siteList.append(site[0])
+
+
+# generate the string of site ids for the url
+idInputString = ",".join(siteList)
+# print(idInputString)
+
+url = "https://waterservices.usgs.gov/nwis/site/?format=rdb&sites=" + idInputString + "&siteStatus=all"
+
+# get data from url
+req = requests.get(url)
+
+# Discard preamble from response
+data = req.text[req.text.rindex('#') + 1:]
+data = data[data.index('\n') + 1:]
+# the first line of data is now the column headers
+
+# turn data into list by splitting on \n
+data = data.split('\n')
+
+# discard 2nd line
+data = [data[0]] + data[2:]
+
+# make a nice dictionary from data
+fields = data[0].split('\t')
+data_nice = []
+for line in data[1:]:
+    line = line.split('\t')
+    line_dict = dict(zip(fields, line))
+    data_nice.append(line_dict)
+
+# keep only station name, id, huc, lat and lon
+arr = []
+
+# store this data in arr
+for datum in data_nice:
```

Comment:
Is there a reason for changing the data into an array of arrays? Would it be clearer to convert the array of dictionaries into a geojson rather than converting to an array of arrays?